### PR TITLE
docs: Add Redpanda Cloud to the quickstart

### LINKED ISSUE #1208: Create Cloud Quickstart doc
### Describe the Issue

We need to offer users a quick and easy onboarding experience wrt Cloud. If browsing our navtree, it is unlikely they will look under Deploy for a cloud QS. Our solution:
- Add a **Cloud** tab in the current [Redpanda QS](https://docs.redpanda.com/docs/get-started/quick-start/quick-start-docker/) 
- Add an entry in the navtree (Deploy > Redpanda Cloud > Redpanda Cloud Quickstart) that links to the Cloud tab in the current RP Quickstart. This way, customers will find the RP Cloud QS whether they are doing an organic search or browsing our navtree.

There is already **some** content in https://docs.redpanda.com/docs/get-started/code-examples/ for getting Cloud up and running. However, this doc is problematic:  the section where it discusses [Create a Producer](https://docs.redpanda.com/docs/get-started/code-examples/#create-producer), cloud users must go back and copy code snippet from previous section and paste it in the code snippet here (I will file a ticket against this topic to make every step includes guidance for cloud users). Still, this material can be leveraged for the cloud QS content. 
-------------------
For an example of a Cloud QS being incorporated into a general QS, see the Cockroach doc (@rkruze will like this :-) ):
https://www.cockroachlabs.com/docs/cockroachcloud/quickstart.html
The cloud material is in a tab here, and this is what we want for our cloud customers.

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: February 21

Fixes #1208

- Added linkable tabs to the quickstart for Redpanda Cloud and Docker.
- Single-sourced content for creating a namespace in Cloud.
- Renamed file from `quick-start-docker` to `quick-start` and added necessary redirects/link updates.
- [Fixed a bug in the version fetcher](https://github.com/redpanda-data/documentation/pull/1260/commits/0f026aeef1b6fa91678362b9e2a6ea0ebda63fa0) where the build failed if the API rate limit was reached. (I came across this when I hit the limit while doing too many local builds 🙈 )
- Added a link to the quickstart from the Cloud deployment section in the nav tree.
- Simplified the Docker Compose samples by using the `--mode dev container` flag.

### CODE REVIEW

File: docs/get-started/quick-start.mdx [hunk 1]
```diff
@@ -0,0 +1,775 @@
+---
+title: Redpanda Quickstart
+---
+
+<head>
+    <meta name="title" content="Redpanda Quickstart | Redpanda Docs"/>
+    <meta name="description" content="Spin up a Redpanda cluster, create a basic streaming application, and explore your cluster in Redpanda Console."/>
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:7]:
```suggestion
    <meta name="description" content="Spin up a Redpanda cluster with Docker or Redpanda Cloud, create a basic streaming application, and explore your cluster in Redpanda Console."/>
```

File: docs/get-started/quick-start.mdx [hunk 2]
```diff
@@ -0,0 +1,775 @@
...
+  - [Kubernetes](../../deploy/deployment-option/self-hosted/kubernetes)
+
+  :::
+
+## Prerequisites
+
+<Tabs groupId="quickstart" queryString>
+<TabItem value="cloud" label="Redpanda Cloud">
+
+[Sign up for a trial](https://redpanda.com/try-redpanda?section=cloud-trial) of Redpanda Cloud Dedicated.
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:39]: The cloud-trial page has the "Redpanda Cloud" tab and a description for "Dedicated Clusters." Should this description call out those specific names?

```suggestion
[Sign up for a trial](https://redpanda.com/try-redpanda?section=cloud-trial) of Redpanda Cloud - Dedicated Clusters.
```

File: docs/get-started/quick-start.mdx [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+
+</TabItem>
+<TabItem value="docker" label="Docker">
+
+You need Docker Compose. For installation instructions, see the [Docker Compose documentation](https://docs.docker.com/compose/install/).
+
+To check if you have Docker Compose, run the following:
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:46]:
```suggestion
To check if you have Docker Compose installed, run the following:
```

File: docs/get-started/quick-start.mdx [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+
+To deploy a Redpanda cluster in the Dedicated cloud, create a namespace, then follow the steps for creating a Dedicated cluster.
+
+To create a namespace:
+
+<CreateNamespace/>
+
+To create a cluster:
+
+1. Click **Create Dedicated cluster**, and enter the following values:
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:70]:
```suggestion
1. Click **Add Cluster**.

1. Click **Create Dedicated cluster**, and enter the following values:
```

File: docs/get-started/quick-start.mdx [hunk 5]
```diff
@@ -0,0 +1,775 @@
...
+
+    - **Cluster name**:
+
+      ```
+      Redpanda quickstart
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:75]: The UI calls for lower case, no whitespace.

```suggestion
      ```
      redpandaquickstart
```

File: docs/get-started/quick-start.mdx [hunk 6]
```diff
@@ -0,0 +1,775 @@
...
+      ```
+
+    - **Throughput**: 1 MBps
+
+    - **Version**: The latest available version
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:80]: Should we differentiate between exact selectable options (Throughput: 1 MBps) vs general options (The latest available version)? A reasonable user will understand after not finding literally "The latest available version", but can we help them, say by bracketing this:

```suggestion
    - **Version**: <The latest available version>
```

File: docs/get-started/quick-start.mdx [hunk 7]
```diff
@@ -0,0 +1,775 @@
...
+<TabItem value="cloud" label="Redpanda Cloud">
+
+In Redpanda Cloud, go to **Topics** > **twitch_chat**.
+
+The message that you produced to the topic is displayed along with some other details about the topic.
+
+</TabItem>
+<TabItem value="docker" label="Docker">
+
+To explore your topic, you can use Redpanda Console, which you deployed in the `docker-compose.yml` file.
```
Comment (shorgi) [inline docs/get-started/quick-start.mdx:704]:
```suggestion
To explore your topic, you can use Redpanda Console, which you deployed in the `docker-compose.yml` file and is running locally with port 8080. 
```

**Review by shorgi — COMMENTED** [review]
See comments for 23.1, I assume they also apply to 22.3

File: docs/get-started/quick-start.mdx (hunk 6 above)
Comment (JakeSCahill) [reply docs/get-started/quick-start.mdx:82]: We could say 'Choose the latest available version'?

Comment (shorgi) [reply docs/get-started/quick-start.mdx]: Sounds good!

**Review by shorgi — APPROVED** [review]
LGTM

File: docs/manage/data-migration.mdx [hunk 8]
```diff
@@ -25,7 +25,7 @@ For details on MirrorMaker 2 and its options, see the [Kafka documentation](http
 
 To set up the replication, you need:
 
-- 2 Redpanda clusters - You can set up these clusters in any deployment you like, including [Kubernetes](../../deploy/deployment-option/self-hosted/kubernetes/get-started-dev), [Docker](../../get-started/quick-start/quick-start-docker), and [Linux](../../deploy/deployment-option/self-hosted/manual).
+- 2 Redpanda clusters - You can set up these clusters in any deployment you like, including [Kubernetes](../../deploy/deployment-option/self-hosted/kubernetes/get-started-dev), [Docker](../../get-started/quick-start), and [Linux](../../deploy/deployment-option/self-hosted/manual).
```
Comment (c4milo) [inline docs/manage/data-migration.mdx:28]: Are there any restrictions on the Redpanda version? do the 2 clusters have to match the version, or is it OK for them to be different versions?

**Comment by c4milo** [comment]
can this be reviewed rendered somewhere? are we using Vercel to host the docs?

**Comment by JakeSCahill** [comment]
@c4milo we're only changing the `/docs/get-started/quick-start` page. You can view it here: https://deploy-preview-1260--redpanda-documentation.netlify.app/docs/get-started/quick-start

**Review by c4milo — APPROVED** [review]

**Pull request merged**